csharp/src/Google.Protobuf/MessageExtensions.cs

```diff
@@ -294,6 +303,16 @@ internal static void MergeFrom(this IMessage message, ReadOnlySequence<byte> dat
             ParsingPrimitivesMessages.CheckReadEndOfStreamTag(ref ctx.state);
         }
 
+        [SecuritySafeCritical]
+        internal static void MergeFrom(this IMessage message, ref ReadOnlySpan<byte> data, bool discardUnknownFields, ExtensionRegistry registry)
```

Comment:
Does data need to be ref here?

Reply:
I think you're right, I removed passing `ReadOnlySpan` by ref in these methods and in `ParseContext`...there are more throughout the codebase, e.g. in `ParsingPrimitives` that probably should also change, but I didn't do that here.

I believe `ReadOnlySpan` is not a big enough struct to benefit from any potential performance increase from passing by `ref` or `in` (which is probably why the Microsoft APIs using it don't use `in`)?

https://docs.microsoft.com/en-us/dotnet/csharp/write-safe-efficient-code#apply-the-in-modifier-to-readonly-struct-parameters-larger-than-systemintptrsize